### PR TITLE
[C-2907] Add contextual-menu, refactor release-date-field

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Adds contextual-menu, using combined pattern from modal-field and mobile's contextual-menu
Removes "submenu" naming
Updates Text, Icon, Modal, and Tile components to work with upload field requirements
Reimplements  ReleaseDateField using new component

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2907-finish-contextual-menus

File: packages/web/src/components/data-entry/ContextualMenu.module.css [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  gap: var(--unit-2);
+  padding: var(--unit-4) var(--unit-6);
+  flex-direction: column;
+}
+
+.preview {
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: var(--unit-2);
```
Comment (amendelsohn) [inline packages/web/src/components/data-entry/ContextualMenu.module.css:11]: `<div classNames={cn(layoutStyles.col, layoutStyles.gap2)} />` will do this

File: packages/web/src/components/data-entry/ContextualMenu.module.css [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  flex-direction: column;
+  gap: var(--unit-2);
+}
+
+.title {
+  display: inline-flex;
+  flex: 1;
+  flex-grow: 1;
+  justify-content: space-between;
+  align-items: center;
```
Comment (amendelsohn) [inline packages/web/src/components/data-entry/ContextualMenu.module.css:25]: `layoutStyles.row` will do this, maybe still need the flex-grow

File: packages/web/src/components/data-entry/ContextualMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    return (
+      <div className={styles.value}>
+        {values?.map((value) => (
+          <SelectedValue key={value} label={value} />
+        ))}
+      </div>
+    )
+  }, [])
+
+  const renderValue = renderValueProp ?? defaultRenderValue
```
Comment (amendelsohn) [inline packages/web/src/components/data-entry/ContextualMenu.tsx:147]: nice! I like this pattern

File: packages/web/src/components/tile/Tile.module.css [hunk 4]
```diff
@@ -15,6 +15,17 @@
...
 }
 
+.flat:hover {
+  border: 1px solid var(--border-strong);
+  background: var(--neutral-light-9);
+}
+
+.flat:active {
+  border-radius: 8px;
+  border: 1px solid var(--border-strong, #e7e6eb);
```
Comment (amendelsohn) [inline packages/web/src/components/tile/Tile.module.css:25]: are we using these backup styles?

File: packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+  return (
+    <ContextualMenu
+      value={value}
+      label={messages.title}
+      description={messages.description}
+      icon={<IconCalendar className={styles.titleIcon} />}
+      initialValues={initialValues}
+      onSubmit={onSubmit}
+      menuForm={
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx:69]: wondering if this should be `menuFields`? You don't need to (and shouldn't) pass a Formik instance into this

File: packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+const RELEASE_DATE = 'releaseDate'
+
+export type ReleaseDateFormValues = {
+  [RELEASE_DATE]: moment.Moment
+}
+
+type ReleaseDateValue = EditFormValues[typeof RELEASE_DATE]
+
+export const ReleaseDateField = () => {
+  const [{ value }, , { setValue }] = useField<ReleaseDateValue>(RELEASE_DATE)
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx:33]: you might want to merge in my multi-track changes before reimplementing the rest of the forms. I had to change a lot of the field, initialValues, and onSubmits

File: packages/web/src/pages/upload-page/components/TrackModalArray.tsx [hunk 7]
```diff
@@ -9,7 +9,7 @@ import styles from './TrackModalArray.module.css'
 export const TrackModalArray = () => {
   return (
     <div className={styles.root}>
-      <ReleaseDateModalForm />
+      <ReleaseDateField />
```
Comment (amendelsohn) [inline packages/web/src/pages/upload-page/components/TrackModalArray.tsx:12]: will the rest of these be renamed or kept as "form"? I see how ReleaseDate is basically a fancy field, but the others all have multiple fields in the modal

**Review by amendelsohn — APPROVED** [review]
Nice work! I think I like the new form pattern, though I'm not sure it saves us all that much in terms of reused code. It's a win in terms of consistency, so probably worthwhile.

File: packages/web/src/components/data-entry/ContextualMenu.module.css (hunk 1 above)
Comment (dylanjeffers) [reply packages/web/src/components/data-entry/ContextualMenu.module.css:11]: oh rightt, hell yeah

File: packages/web/src/components/tile/Tile.module.css (hunk 4 above)
Comment (dylanjeffers) [reply packages/web/src/components/tile/Tile.module.css:25]: no excellent catch

File: packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx (hunk 5 above)
Comment (dylanjeffers) [reply packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx:69]: yeah great call! cause its not the form itself

File: packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx (hunk 6 above)
Comment (dylanjeffers) [reply packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx:33]: yeah was thinking this when i saw the merge conflict, was like whew glad we paused here

File: packages/web/src/pages/upload-page/components/TrackModalArray.tsx (hunk 7 above)
Comment (dylanjeffers) [reply packages/web/src/pages/upload-page/components/TrackModalArray.tsx:12]: yeah that's exactly what im thinking, like at the end of the day these are fields that happen to render a contextual-menu that happens to render a form :D

**Comment by dylanjeffers** [comment]
> Nice work! I think I like the new form pattern, though I'm not sure it saves us all that much in terms of reused code. It's a win in terms of consistency, so probably worthwhile.

Thanks! I think what helps is the saved code with the previews and formik instantition on each modal field, but yeah I basically borrowed everything you wrote for it

**Pull request merged**